Add simple REST switch

bachp opened the pull request:
The switch can get the state via GET and set the state via POST
on a given REST resource.
It is not able to control arbitrary urls but it allows controlling
switches exposed via "real" REST.

maddox commented on homeassistant/components/switch/rest.py:
```diff
@@ -0,0 +1,114 @@
+*Optional
+The body that represents enabled state. Default is "ON".
+
+payload_off
```
should this be `body_on`?

bachp replied:
Yes it should be. I first had it named payload like the mqtt switch, but body made more sense in a REST context.
Thanks for pointing it out.

bachp commented:
PR updated based on feedback from @maddox

balloob commented on homeassistant/components/switch/rest.py:
```diff
@@ -0,0 +1,114 @@
+    if config.get('resource') is None:
+        _LOGGER.error("Missing required variable: resource")
+        return False
+
```
Please add a check if the resource exists and error out if it doesn't.

happyleavesaoc commented on homeassistant/components/switch/rest.py:
```diff
@@ -0,0 +1,114 @@
+        if request.status_code == 200:
+            self._state = False
+        else:
+            _LOGGER.error("Can't turn off pin %s. Is device offline?",
```
s/pin/resource/?

bachp commented:
PR updated based on feedback from:
@happyleavesaoc: Log error for turn_off aligned with turn_on
@balloob: Check added if resource exists like in arest.py

fabaff commented on homeassistant/components/switch/rest.py:
```diff
@@ -0,0 +1,127 @@
+  name: "Bedroom Switch"
+  resource: "http://IP_ADDRESS/ENDPOINT"
+  body_on: "ON"
+  body_off: "OFF"
```
I would suggest that we stick with the terminology used by requests and replace `body` with `payload`

maddox replied:
People using this are going to be more familiar with REST and HTTP terms than they are with the Requests python library. It should probably match the context of the component, not the underlying library.

bachp replied:
I agree that the context is more important then the underlying library, for all we know the implementation can change but the configuration should not need to be changed.

balloob commented:
It looks like you have [1 style violation](https://travis-ci.org/balloob/home-assistant/builds/86479167#L295) that should be fixed. For the rest it looks good to go!

bachp committed:
Add simple REST switch

The switch can get the state via GET and set the state via POST
on a given REST resource.
It is not able to control arbitrary urls but it allows controlling
switches exposed via "real" REST.

bachp commented:
Pull request updated
@balloob The violation is gone now

fabaff commented:
This is the [doc page](https://github.com/balloob/home-assistant.io/blob/master/source/components/switch.rest.markdown) for the switch. Please feel free to add add further details.
